File: FizzBuzz.java
// This is a comment.

public class FizzBuzz {

    private int max_;

    //constructor
    public FizzBuzz(int max) {
        this.max_ = max;
    }

    public void run_() {
        for (int i = 1; i <= this.max_; ++i) {
            printNum_(i);
        }
    }

    public void printNum_(int i) {
        if (i % 15 == 0) {
            System.out.println("FizzBuzz");
        } else if (i % 3 == 0) {
            System.out.println("Fizz");
        } else if (i % 5 == 0) {
            System.out.println("Buzz");
        } else {
            System.out.println(i);
        }
    }

    public static void main(String[] args) {
        FizzBuzz fb = new FizzBuzz(100);
        fb.run_();
    }

}

File: 028_Objective-C/fizzbuzz.m
// This is a comment.

#import <Foundation/Foundation.h>

@interface FizzBuzz: NSObject {
    int max;
}
- (void)run;
- (char *)getValue:(int)i;
- (void)setMax:(int)v;
@end

@implementation FizzBuzz
- (void)setMax:(int)v {
    max = v;
}
- (void)run {
    for (int i = 1; i <= max; ++i) {
        printf("%s\n", [self getValue: i]); //`[self getValue: i]` (Objective-C) is equivalent to `self.getValue(i)` (pseudo-language).
    }
}
- (char *)getValue:(int)i {
    static char buf[256];
    if (i % 3 == 0 && i % 5 == 0) {
        return "FizzBuzz";
    } else if (i % 3 == 0) {
        return "Fizz";
    } else if (i % 5 == 0) {
        return "Buzz";
    } else {
        sprintf(buf, "%d", i);
        return buf;
    }
}
@end

int main() {
    id fb = [FizzBuzz new]; //`id` is a pointer type which points to an object of the type `objc_object`.
    [fb setMax: 100];
    [fb run];
}

